sky/skylet/providers/aws/node_provider.py

```diff
@@ -294,8 +293,25 @@ def create_node(self, node_config, tags, count) -> Dict[str, Any]:
                         "Values": [tags[TAG_RAY_USER_NODE_TYPE]],
                     }
                 )
+            filters_with_launch_config = copy.copy(filters)
+            filters_with_launch_config.append({
+                    "Name": "tag:{}".format(TAG_RAY_LAUNCH_CONFIG),
+                    "Values": [tags[TAG_RAY_LAUNCH_CONFIG]],
+                })
+
+            def _filter_nodes(filters, priortized_nodes=[]):
+                nodes = list(self.ec2.instances.filter(Filters=filters))
+                if priortized_nodes:
+                    priortized_node_ids = {n.id for n in priortized_nodes}
+                    nodes.sort(key=lambda n: n.id in priortized_node_ids, reverse=True)
+                return nodes[:count]
+
+            # SkyPilot: Try to reuse nodes with the same launch config first,
+            # then fall back to nodes with any launch config.
```

Comment:
Update to? SkyPilot: We try to use the instances with the same matching launch_config first. If there is not enough instances with matching launch_config, we then use all the instances with the same matching launch_config plus some instances with wrong launch_config.